website/docs/plugin/terraform-plugin-protocol.mdx

```diff
@@ -0,0 +1,227 @@
+* [tf5to6server](/plugin/mux/translating-protocol-version-5-to-6): A package to translate protocol version 5 providers into protocol version 6.
+* [tf6muxserver](/plugin/mux/combining-protocol-version-6-providers): A package to combine multiple protocol version 6 providers.
+
+## Protocol Version 5
+
+Protocol version 5 is compatible with Terraform CLI version 0.12 and later.
+
+Implementations include:
+
+* [terraform-plugin-framework](/plugin/framework): A higher-level SDK that makes Terraform provider development easier by abstracting implementation details.
+* [terraform-plugin-sdk/v2](/plugin/sdkv2): A higher-level SDK that makes Terraform provider development easier by abstracting implementation details.
+* [terraform-plugin-go tf5server](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server): A lower-level SDK to develop Terraform providers for more advanced use cases.
+* [tf6to5server](/plugin/mux/translating-protocol-version-6-to-5): A package to translate protocol version 6 providers into protocol version 5.
+* [tf5muxserver](/plugin/mux/combining-protocol-version-5-providers): A package to combine multiple protocol version 5 providers.
+
+## Protocol RPCs
+
+The Terraform plugin protocol specifies the following RPCs:
+
+| Protocol Version 5           | Protocol Version 6           |
+|------------------------------|------------------------------|
+| `GetSchema`                  | `GetProviderSchema`          |
+| `PrepareProviderConfig`      | `ValidateProviderConfig`     |
+| `ValidateResourceTypeConfig` | `ValidateResourceConfig`     |
+| `ValidateDataSourceConfig`   | `ValidateDataResourceConfig` |
+| `UpgradeResourceState`       | `UpgradeResourceState`       |
+| `Configure`                  | `ConfigureProvider`          |
+| `ReadResource`               | `ReadResource`               |
+| `ReadDataSource`             | `ReadDataSource`             |
+| `PlanResourceChange`         | `PlanResourceChange`         |
+| `ApplyResourceChange`        | `ApplyResourceChange`        |
+| `ImportResourceState`        | `ImportResourceState`        |
+
+![diagram: Terraform Plugin Protocol 6 RPCs](/img/docs/terraform-plugin-protocol-6-rpcs-all.png)
+
+## RPCs and Terraform commands
+
+The following illustrate the RPCs that are sent during execution of the specified Terraform commands.
+
+The RPCs issued are for a provider that is using version 6 of the Terraform plugin protocol when the Terraform configuration contains configuration for the provider, a resource and a data source.
+
+### _terraform validate_
+
+Executing _terraform validate_ results in the following unique RPCs:
+
+![diagram: Terraform Plugin Protocol 6 Validate RPCs](/img/docs/terraform-plugin-protocol-6-rpcs-validate.png)
+
+The actual sequence of RPCs is as follows:
+
+- [GetProviderSchema](#getproviderschema-rpc)
+- [GetProviderSchema](#getproviderschema-rpc)
```

Comment:
The duplicate calls to the RPC are an implementation detail of how Terraform currently works and not part of any contract with providers. I'd suggest removing duplicates here and below.